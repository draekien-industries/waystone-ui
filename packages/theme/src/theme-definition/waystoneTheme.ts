import type { Theme } from 'theme-ui';
import { colors } from './colors';
import { breakpoints } from './breakpoints';
import { typography } from './typography';
import { layouts } from './layouts';
import { variants } from './variants';
import { styles } from './styles';

export const waystoneTheme = {
  config: {
    useLocalStorage: true,
    useCustomProperties: true,
    useRootStyles: true,
    useBorderBox: true,
    useColorSchemeMediaQuery: true,
  },
  colors,
  breakpoints: [...breakpoints],
  ...typography,
  ...layouts,
  ...styles,
  ...variants,
  styles: {
    root: {
      fontFamily: 'body',
      fontWeight: 'regular',
      fontSize: '16px',
      lineHeight: 'body',
      transition: 'all 200ms ease',
    },
  },
} as const satisfies Theme;

export type WaystoneTheme = typeof waystoneTheme;
