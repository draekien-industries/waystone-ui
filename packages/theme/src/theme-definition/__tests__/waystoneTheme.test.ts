import { waystoneTheme } from '../waystoneTheme';

describe('waystoneTheme', () => {
  it('should match snapshot', () => {
    expect(waystoneTheme).toMatchSnapshot();
  });
});
