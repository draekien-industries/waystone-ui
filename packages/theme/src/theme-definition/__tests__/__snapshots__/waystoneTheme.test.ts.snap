// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`waystoneTheme should match snapshot 1`] = `
{
  "breakpoints": [
    "480px",
    "768px",
    "1024px",
    "1200px",
  ],
  "colors": {
    "a-100": "#dbf5ff",
    "a-200": "#ade8ff",
    "a-300": "#7ad9ff",
    "a-400": "#33c5ff",
    "a-50": "#ebf9ff",
    "a-500": "#00a8eb",
    "a-600": "#009ad6",
    "a-700": "#0087bd",
    "a-800": "#00719e",
    "a-900": "#005070",
    "a-950": "#003347",
    "accent": "#00a9ed",
    "active": "white",
    "b-100": "#ebebeb",
    "b-200": "#d4d4d4",
    "b-300": "#b8b8b8",
    "b-400": "#969696",
    "b-50": "#f5f5f5",
    "b-500": "#666666",
    "b-600": "#5e5e5e",
    "b-700": "#4f4f4f",
    "b-800": "#404040",
    "b-900": "#303030",
    "b-950": "#1a1a1a",
    "background": "#f4f9fa",
    "card": "white",
    "code": "#4f4f4f",
    "error-100": "#ffe5ea",
    "error-200": "#ffc7d0",
    "error-300": "#ffa3b3",
    "error-400": "#ff6b84",
    "error-50": "#fff5f7",
    "error-500": "#e00025",
    "error-600": "#cc0022",
    "error-700": "#b8001f",
    "error-800": "#8f0018",
    "error-900": "#660011",
    "error-950": "#57000e",
    "ghost": "#ebebeb",
    "h-100": "#fdede3",
    "h-200": "#fad7c2",
    "h-300": "#f7bc97",
    "h-400": "#f39d68",
    "h-50": "#fef6f1",
    "h-500": "#ed6f21",
    "h-600": "#de6012",
    "h-700": "#c15410",
    "h-800": "#a0460d",
    "h-900": "#76330a",
    "h-950": "#502307",
    "highlight": "#ed6f22",
    "info-100": "#d3effd",
    "info-200": "#9ddcfb",
    "info-300": "#53c1f8",
    "info-400": "#099de7",
    "info-50": "#ebf8fe",
    "info-500": "#066798",
    "info-600": "#055d89",
    "info-700": "#055076",
    "info-800": "#044362",
    "info-900": "#022b40",
    "info-950": "#021b27",
    "modes": {
      "dark": {
        "active": "#5e5e5e",
        "background": "#1a1a1a",
        "card": "#404040",
        "code": "#ed6f22",
        "ghost": "#404040",
        "muted": "#5e5e5e",
        "text": "#c6dfe7",
      },
    },
    "muted": "#b8b8b8",
    "p-100": "#e2eff3",
    "p-200": "#c6dfe7",
    "p-300": "#a5ccd9",
    "p-400": "#7ab4c7",
    "p-50": "#f4f9fa",
    "p-500": "#4893ac",
    "p-600": "#42879e",
    "p-700": "#3b788c",
    "p-800": "#2f5f6f",
    "p-900": "#21444f",
    "p-950": "#1a343d",
    "primary": "#4995ae",
    "s-100": "#f4eef6",
    "s-200": "#e8daec",
    "s-300": "#dbc5e2",
    "s-400": "#cdaed6",
    "s-50": "#f9f5fa",
    "s-500": "#ba8fc7",
    "s-600": "#af7ebe",
    "s-700": "#a26ab4",
    "s-800": "#8d509f",
    "s-900": "#663a73",
    "s-950": "#42264b",
    "secondary": "#ba90c7",
    "success-100": "#d8f3e1",
    "success-200": "#a9e5bd",
    "success-300": "#6ed390",
    "success-400": "#38b863",
    "success-50": "#ebf9f0",
    "success-500": "#2b8d4b",
    "success-600": "#267d43",
    "success-700": "#216d3b",
    "success-800": "#1b5a30",
    "success-900": "#144224",
    "success-950": "#0d2b17",
    "text": "#1a1a1a",
    "text-disabled": "#969696",
    "warning-100": "#fce8e3",
    "warning-200": "#fad1c7",
    "warning-300": "#f6af9d",
    "warning-400": "#f18569",
    "warning-50": "#fef4f1",
    "warning-500": "#ea441a",
    "warning-600": "#d23913",
    "warning-700": "#bb3311",
    "warning-800": "#9f2b0e",
    "warning-900": "#6c1d0a",
    "warning-950": "#4b1407",
  },
  "config": {
    "useBorderBox": true,
    "useColorSchemeMediaQuery": true,
    "useCustomProperties": true,
    "useLocalStorage": true,
    "useRootStyles": true,
  },
  "fontSizes": {
    "body": "1rem",
    "caption": "0.875rem",
    "heading": "1.75rem",
    "hero": "2.5rem",
    "small": "0.75rem",
    "subtitle": "1.25rem",
    "title": "1.5rem",
  },
  "fontWeights": {
    "bold": 700,
    "medium": 500,
    "regular": 400,
    "semibold": 600,
    "thin": 300,
  },
  "fonts": {
    "body": ""Barlow", sans-serif",
    "heading": ""Barlow", sans-serif",
    "monospace": ""Fira Code", monospace",
  },
  "letterSpacings": {
    "body": "normal",
    "uppercase": "0.1em",
  },
  "lineHeights": {
    "body": 1.5,
    "heading": 1.125,
  },
  "radii": {
    "extra-large": "1rem",
    "large": "0.75rem",
    "max": 999,
    "medium": "0.5rem",
    "small": "0.25rem",
  },
  "shadows": {
    "large": "inset 0 0 0.5px 1px hsla(0, 0%, 100%, 0.075), 0 0 8px 4px hsla(0, 0%, 0%, 0.05), 0 1.2px 1.6px hsla(0, 0%, 0%, 0.02), 0 3.6px 6px hsla(0, 0%, 0%, 0.045), 0 14px 24px hsla(0, 0%, 0%, 0.09)",
    "medium": "inset 0 0 0.5px 1px hsla(0, 0%, 100%, 0.075), 0 0 4px 2px hsla(0, 0%, 0%, 0.05), 0 0.6px 0.8px hsla(0, 0%, 0%, 0.02), 0 1.8px 3px hsla(0, 0%, 0%, 0.045), 0 7px 12px hsla(0, 0%, 0%, 0.09)",
    "none": "none",
    "small": "inset 0 0 0.5px 1px hsla(0, 0%, 100%, 0.075), 0 0 0 1px hsla(0, 0%, 0%, 0.05), 0 0.3px 0.4px hsla(0, 0%, 0%, 0.02), 0 0.9px 1.5px hsla(0, 0%, 0%, 0.045), 0 3.5px 6px hsla(0, 0%, 0%, 0.09)",
  },
  "sizes": {
    "extra-large": "20em",
    "extra-small": "1em",
    "full": "100%",
    "half": "50%",
    "large": "15em",
    "medium": "10em",
    "small": "5em",
  },
  "space": {
    "extra-large": "2rem",
    "extra-small": "0.25rem",
    "large": "1.5rem",
    "medium": "1rem",
    "small": "0.5rem",
  },
  "styles": {
    "root": {
      "fontFamily": "body",
      "fontSize": "16px",
      "fontWeight": "regular",
      "lineHeight": "body",
      "transition": "all 200ms ease",
    },
  },
  "text": {
    "body": {
      "fontFamily": "body",
      "fontSize": "body",
      "fontWeight": "regular",
      "lineHeight": "body",
      "marginY": "small",
    },
    "caption": {
      "fontFamily": "body",
      "fontSize": "caption",
      "fontWeight": "regular",
      "lineHeight": "body",
    },
    "default": {
      "fontFamily": "body",
      "fontSize": "body",
      "fontWeight": "regular",
      "lineHeight": "body",
      "marginY": "small",
    },
    "eyebrow": {
      "fontFamily": "body",
      "fontSize": "body",
      "fontWeight": "medium",
      "letterSpacing": "uppercase",
      "lineHeight": "body",
      "textTransform": "uppercase",
    },
    "heading": {
      "fontFamily": "heading",
      "fontSize": "heading",
      "fontWeight": "bold",
      "lineHeight": "heading",
      "marginY": "medium",
    },
    "hero": {
      "fontFamily": "heading",
      "fontSize": "hero",
      "fontWeight": "bold",
      "lineHeight": "heading",
      "marginY": "large",
    },
    "label": {
      "fontFamily": "body",
      "fontSize": "body",
      "fontWeight": "medium",
      "lineHeight": "body",
      "marginY": 0,
    },
    "monospace": {
      "fontFamily": "monospace",
      "fontSize": "body",
      "fontWeight": "regular",
      "lineHeight": "body",
    },
    "small": {
      "fontFamily": "body",
      "fontSize": "small",
      "fontWeight": "regular",
      "lineHeight": "body",
    },
    "subtitle": {
      "fontFamily": "body",
      "fontSize": "subtitle",
      "fontWeight": "semibold",
      "lineHeight": "heading",
      "marginY": "small",
    },
    "title": {
      "fontFamily": "heading",
      "fontSize": "title",
      "fontWeight": "bold",
      "lineHeight": "heading",
      "marginY": "small",
    },
  },
  "zIndices": {
    "drawer": 40,
    "dropdown": 30,
    "high": 40,
    "higher": 50,
    "highest": 60,
    "low": 20,
    "lower": 10,
    "lowest": 0,
    "middle": 30,
    "modal": 50,
    "navbar": 20,
    "pagination": 1,
    "toast": 60,
  },
}
`;
